### PR TITLE
First pass at changelog for 1.0

### PR DESCRIPTION
**Pull request by evan2645**
Building this changelog was fairly difficult as we thought we were going to release 1.0 soon back in December ... as such, we began accumulating changes earmarked for 1.0 in the main branch quite some time ago. Many months worth of cherry-picked releases made digging up the unreleased changes in main fairly difficult. Please be thorough in reviewing this update as I'm fairly certain that there are some things missing and possibly some duplication.

I've also gone ahead and added the 0.12.3 changelog entry which was missing.

Signed-off-by: Evan Gilman <egilman@vmware.com>

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,63 @@
...
+- Workload API JWT-SVID profile will no longer return non-SPIFFE claims on validated JWTs from foreign trust domains (#2372)
+- The SPIFFE bundle endpoint TLS stack now supports a smaller set of signature schemes (#2397)
+- SPIRE artifact tarball no longer extracts `.` to avoid inadvertent changes in directory permisions (#2219)
+- SPIRE Server default socket path is now `/tmp/spire-server/private/api.sock` (#2075)
+- SPIRE Agent default socket path is now `/tmp/spire-agent/public/api.sock` (#2075)
+
+### Deprecated
+- SPIRE Server federation configuration in the `federates_with` `bundle_endpoint` block is now deprecated (#2340)
+- SPIRE Server `gcp_iit` NodeAttestor configurable `projectid_whitelist` is deprecated in favor of `projectid_allow_list`
+- SPIRE Server `k8s_sat` and `k8s_psat` NodeAttestor configurable `service_account_whitelist` is deprecated in favor of `service_account_allow_list`
```
Comment (evan2645) [inline CHANGELOG.md:30]: This line and the one above should are missing PR number

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,63 @@
 # Changelog
 
+## [1.0.0] - 2021-07-08
+
+### Added
+- The `vault` UpstreamAuthority plugin now supports Kubernetes service account authentication (#2356)
+- A new `cert-manager` UpstreamAuthority plugin is now available (#2274)
+- SPIRE Server CLI can now be used to ban agents (#2374)
+- SPIRE Server CLI now has `count` subcommands for agents, entries, and bundles (#2128)
+- SPIRE Server can now be configured for SPIFFE federation using the `https_web` and `https_spiffe` profiles (#2340)
```
Comment (amartinezfayo) [inline CHANGELOG.md:10]: The way that this is worded may be a little confusing, I think. The `https_web` and `https_spiffe` profiles were supported previously also, but the configuration was not explicit about using those names. So maybe I would reword this to reflect the fact that the configuration format for federation relationships was changed to be more consistent with the language expressed in the SPIFFE Federation spec.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,63 @@
...
+- SPIRE Agent may now optionally provided unregistered callers with a bundle for SVID validation via the `allow_unauthenticated_verifiers` configurable (#2102)
+- SPIRE Server JWT key type is now independently configurable via `jwt_key_type` (#1991)
+
+### Changed
+- SPIRE Server's SVID now uses the key type configured as `ca_key_type` (#2269)
+- Caller address is now logged for agent API calls resulting in an error (#2281)
+- Agent SVID renewals are now logged by the server at the INFO level (#2309)
+- Workload API JWT-SVID profile will now return an error if the caller is unidentified (#2369)
+- Workload API JWT-SVID profile will no longer return non-SPIFFE claims on validated JWTs from foreign trust domains (#2372)
+- The SPIFFE bundle endpoint TLS stack now supports a smaller set of signature schemes (#2397)
```
Comment (evan2645) [inline CHANGELOG.md:22]: It would be good if we can provide some more detail here. Any suggestions @azdagron?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,63 @@
...
+## [1.0.0] - 2021-07-08
+
+### Added
+- The `vault` UpstreamAuthority plugin now supports Kubernetes service account authentication (#2356)
+- A new `cert-manager` UpstreamAuthority plugin is now available (#2274)
+- SPIRE Server CLI can now be used to ban agents (#2374)
+- SPIRE Server CLI now has `count` subcommands for agents, entries, and bundles (#2128)
+- SPIRE Server can now be configured for SPIFFE federation using the `https_web` and `https_spiffe` profiles (#2340)
+- SPIRE Server and Agent now expose the standard gRPC health service (#2057, #2058)
+- SPIFFE bundle endpoint URL is now configurable (#2278)
```
Comment (amartinezfayo) [inline CHANGELOG.md:12]: #2278 is the issue, the PR is #2340. I think that it would be good to make it clear that this is referring to the foreign trust domains when configuring federation relationships, not the SPIFFE bundle endpoint that may be served by _this_ SPIRE Server.

**Review by amartinezfayo — COMMENTED** [review]
This is looking great @evan2645!
I'm thinking that we may also include the following:
- #2062
- #1996
- #2021
- #1967

**Comment by evan2645** [comment]
Here's the list of PRs captured in the changelog for posterity

 #2356
 #2274
 #2374
 #2128
 #2340
 #2057
 #2058
 #2278
 #2102
 #1991
 #2269
 #2281
 #2309
 #2369
 #2372
 #2397
 #2219
 #2075
 #2075
 #2340
 #2075
 #2368
 #2093
 #2098
 #2191
 #2189
 #2161
 #2021
 #2370
 #2058

File: CHANGELOG.md (hunk 3 above)
Comment (azdagron) [reply CHANGELOG.md:22]: Maybe something like `The SPIFFE bundle endpoint supports a reduced signature algorithm set to improve key manager compatibility`?

**Comment by azdagron** [comment]
I think we need also need #2390.

File: CHANGELOG.md (hunk 3 above)
Comment (evan2645) [reply CHANGELOG.md:22]: Oops sorry I meant more detail as in, what exactly is supported or not supported now (as opposed to why). I checked the PR and saw three signature schemes but they were a mouthful :) perhaps there's a different way to say the equivalent thing?

**Review by azdagron — APPROVED** [review]
\o/
